frontend/src/modules/Dashboard/Components/GroupCard.tsx

```diff
@@ -36,20 +39,45 @@ export const GroupCard = ({
     }
     setOpen(false)
   }
-
+  // returns color of background, button, and if newly matchable
+  function getColor(students: number, groups: number) {
+    //all students are matched
+    if (students === 0 && groups > 0) {
+      return [colors.paleviolet, colors.darkpurple, 'no']
+    }
+    //students are ready to be matched
+    else if (newStudents > 0 && groupsFormed > 0) {
+      return [colors.lightgreen, colors.darkgreen, 'no']
+    }
+    //NEWLY MATCHABLE
+    else if (newStudents > 1 && groupsFormed === 0) {
+      return [colors.lightgreen, colors.darkgreen, 'yes']
+    }
+    //only 1 student & 0 groups formed
+    else return [colors.lightyellow, colors.darkyellow, 'no']
+  }
+  const styleArray = getColor(
+    { newStudents }.newStudents,
+    { groupsFormed }.groupsFormed
+  )
```

Comment:
Is there a reason you're using this sort of syntax? They're both numbers, so why can't you just pass them in to the function? like so:
`const styleArray = getColor(newStudents, groupsFormed)`